extensions/mssql/src/dacfx/dacFxService.ts

```diff
@@ -130,4 +130,15 @@ export class DacFxService implements mssql.IDacFxService {
 			}
 		);
 	}
+
+	public parseTSqlScript(filePath: string, databaseSchemaProvider: string): Thenable<mssql.ParseTSqlScriptResult> {
+		const params: contracts.ParseTSqlScriptParams = { filePath, databaseSchemaProvider };
+		return this.client.sendRequest(contracts.ParseTSqlScriptRequest.type, params).then(
```

Comment:
I'd also suggest just making this async and getting rid of the thenable - there isn't any reason for our extension apis to be using thenable.

Reply:
done